### ISSUE TITLE
@root.descendants for page /tag:foo doesn't contain root page and recursively ALL children, but only with tag:foo

### ISSUE THREAD

**Issue by kirillaristov** [getgrav/grav#2365]
This code snippet (https://learn.getgrav.org/content/collections#advanced-collections) doesn't work for pages with /tag:foo on it's url end.

```
{% set options = { items: {'@root.descendants':''}, 'order': {'by': 'folder', 'dir': 'asc'}} %}
{% set my_collection = page.collection(options) %}

{% for p in my_collection %}
{% if p.header.menu %}
    <ul>
    {% if page.slug == p.slug %}
        <li class="{{ p.slug }} active"><span>{{ p.menu }}</span></li>
    {% else %}
        <li class="{{ p.slug }}"><a href="{{ p.url }}">{{ p.menu }}</a></li>
    {% endif %}
    </ul>
{% endif %}
{% endfor %}
```

Steps to reproduce:
1) Put in any directory several pages with taxonomy tag:foo
2) Make template like above for menu generating 
3) Make page with iterate through all page.children and output them (https://kirillaristov.com/en/my-routes/as-the-first-settlers/media)
4) Add tagcloud
5) Url to this directory, e.g.: https://kirillaristov.com/en/my-routes/as-the-first-settlers/media/tag:fav
6) Check menu - it is empty. var_dump for @root.descendants contains only pages with tag:fav

Plese, help. Sorry for the second post, I think the last time I wrote is unclear.

**Comment by mahagr** [comment]
@kirillaristov Can you try to add extra option: `url_taxonomy_filters: false`?

**Comment by kirillaristov** [comment]
> @kirillaristov Can you try to add extra option: `url_taxonomy_filters: false`?

Tried. If I add this parameter, the page content is displayed ignoring tag:foo in the url.

**Comment by mahagr** [comment]
Yes, it was just to confirm the issue to me. So basically you can do either/or, there's no way to filter the tree without removing parent items from the collection.

**Comment by kirillaristov** [comment]
@mahagr Let's be clear, just to be sure. 

I did this for main menu generating

```
{% set options = { items: {'@root.descendants':''}, 'order': {'by': 'folder', 'dir': 'asc'}} %}
{% set my_collection = page.collection(options) %}
```

and then url to example.com/example_gallery/tag:fav_photos

After this @root.descendants from upper code contains only tag:fav collection, *without* root element and other.

So I can request page.collection(options) only one time. Oh, how sad.

**Comment by mahagr** [comment]
Yes, that's how it goes. It's a collection, not a nested sets with parent(s).

**Comment by kirillaristov** [comment]
@mahagr Maybe there is another way to use in one page tag:foo and dynamic menu generating? Now it turns out that if the menu is created dynamically, you can not use tag:foo in urls, which is useful, e.g. for tagcloud.

**Comment by mahagr** [comment]
There is some javascript logic in the learn site for example, which displays all the pages matching to the content. I was just trying to think if there are any methods to do all of this automatically, but I couldn't think of anything. It is possible though, by creating a method which reconstructs a parent tree from a collection, but I am not sure if it has been done before.

**Comment by kirillaristov** [comment]
@mahagr Really, I don't understant - is it not so useful function to have possibility access root and all descendants at any time and any times? I'm confused.

**Comment by mahagr** [comment]
A collection isn't meant to do what you want -- it is just a flat list of objects which match the requirements. It is not meant to be anything else than that, so if you want to build a tree with all the matching items and their parents, you would need another approach to do that.

**Comment by kirillaristov** [comment]
@mahagr 
I call page.collection twice, in two different places.
The first call is in the menu template (standalone collection object!):
```
{% set my_collection = page.collection({ items: {'@root.descendants':''}}) %}

{% for item in my_collection %}
    <li>{{ item.title }}</li>
{% endfor %}
```

The second call in the page content template (default collection):
(page frontmatter)
```
content:
    items: '@self.children'
    order:
        by: default
        dir: desc
    limit: 0
    pagination: false
```
(page content template)
```
{% for item in page.collection %}
    <p>{{ item.title }}</p>
{% endfor %}
```

This design works on all pages of the site, except on pages with /tag:foo on it's url end. On pages with /tag:foo in the url `page.collection` with options `items: {'@root.descendants':''}}` contains only pages with `tag = foo`.

The problem is that you cannot access the root element when the page address contains `/tag:foo`.

This behavior seems incomprehensible to me personally. I have not yet found a way to solve my problem otherwise.
